presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -233,6 +233,7 @@ private void processNewSources()
         long maxRuntime = duration.roundTo(TimeUnit.NANOSECONDS);
 
         Optional<ListenableFuture<?>> result = tryWithLock(100, TimeUnit.MILLISECONDS, () -> {
+            driverContext.getYieldSignal().setWithDelay(maxRuntime, driverContext.getYieldExecutor());
```

Comment:
yield signal will be reused for multiple times; is it proper to make it a `Closeable`?